frostsnap_coordinator/src/signing.rs

```diff
@@ -6,25 +6,28 @@ use frostsnap_core::{
     },
     DeviceId,
 };
-use std::collections::HashSet;
+use std::collections::BTreeSet;
 
 /// Keeps track of when
 #[derive(Debug)]
 pub struct SigningDispatcher {
-    need_to_send_to: HashSet<DeviceId>,
+    need_to_send_to: BTreeSet<DeviceId>,
     signing_state_changed: bool,
     // FIXME: make accessors
     pub request: SignRequest,
     pub finished_signatures: Option<Vec<EncodedSignature>>,
-    pub targets: HashSet<DeviceId>,
-    pub got_signatures: HashSet<DeviceId>,
+    pub targets: BTreeSet<DeviceId>,
+    pub got_signatures: BTreeSet<DeviceId>,
 }
 
 impl SigningDispatcher {
     /// Takes in the messages from `start_sign` and extracts the signing request to handle separately.
     ///
     /// We need to do this because we want to only send out the message to the devices that are connected.
-    pub fn from_filter_out_start_sign(start_sign_messages: &mut Vec<CoordinatorSend>) -> Self {
+    pub fn from_filter_out_start_sign(
+        start_sign_messages: &mut Vec<CoordinatorSend>,
+        targets: BTreeSet<DeviceId>,
```

Comment:
So with my suggestion above I'm saying that this wouldn't be passed in here -- it'll be in the `ToDevice` variant below.